Lombiq.Hosting.Tenants.Maintenance/Maintenance/UpdateOpenIdSettings/CustomMaintenanceProvider.cs

```diff
@@ -0,0 +1,44 @@
+using Lombiq.Hosting.Tenants.Maintenance.Extensions;
+using Lombiq.Hosting.Tenants.Maintenance.Models;
+using Lombiq.Hosting.Tenants.Maintenance.Services;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Lombiq.Hosting.Tenants.Maintenance.Maintenance.UpdateOpenIdSettings;
+
+public class CustomMaintenanceProvider : MaintenanceProviderBase
```

Comment:
I've asked about it on the other PR, but I ask it here, too. Why do you need another abstraction layer for such providers? Couldn't all they be implemented using the `MaintenanceProviderBase` as the base?